msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -3112,4 +3112,17 @@ private boolean isUseAnonymousId(Topic topic)
 		// Condenses to:
 		return topic.getPostAnonymous() && (!topic.getRevealIDsToRoles() || !uiPermissionsManager.isIdentifyAnonAuthors((DiscussionTopic) topic));
 	}
+
+    public int getPercentRead()
+    {
+        int totalPosts = 0;
+        List<Object[]> studentAuthoredStats = messageManager.findAuthoredMessageCountForAllStudents();
+        for (Object[] pair : studentAuthoredStats)
+        {
+            totalPosts += ((Long) pair[1]).intValue();
```

Comment:
this just smells like it can blow up, no?

Reply:
I've tested it for both users that have read some, and those who have read none: no explosions.

It's certainly not how I would have coded `findAuthoredMessageCountForAllStudents()`, but it does seem to gaurantee zeros instead of nulls.

Reply:
ok not sure how but ok !